executor/infoschema_reader.go

```diff
@@ -2174,26 +2180,100 @@ func dataForAnalyzeStatusHelper(sctx sessionctx.Context) (rows [][]types.Datum,
+			if RemainDurationErr != nil {
+				log.Warn("get remaining duration failed", zap.Error(RemainDurationErr))
+			}
+			if RemainingDuration != nil {
+				remainDurationStr = execdetails.FormatDuration(*RemainingDuration)
+			}
+			progressStr = progress
+			estimatedRowCntStr = estimatedRowCnt
+		}
+		var row []types.Datum
+		if isShow {
+			row = types.MakeDatums(
+				dbName,             // TABLE_SCHEMA
+				tableName,          // TABLE_NAME
+				partitionName,      // PARTITION_NAME
+				jobInfo,            // JOB_INFO
+				processedRows,      // ROW_COUNT
+				startTime,          // START_TIME
+				endTime,            // END_TIME
+				state,              // STATE
+				failReason,         // FAIL_REASON
+				instance,           // INSTANCE
+				procID,             // PROCESS_ID
+				remainDurationStr,  // REMAINING_SECONDS
+				progressStr,        // PROGRESS
+				estimatedRowCntStr, // ESTIMATED_TOTAL_ROWS
+			)
+		} else {
+			row = types.MakeDatums(
+				dbName,        // TABLE_SCHEMA
+				tableName,     // TABLE_NAME
+				partitionName, // PARTITION_NAME
+				jobInfo,       // JOB_INFO
+				processedRows, // ROW_COUNT
+				startTime,     // START_TIME
+				endTime,       // END_TIME
+				state,         // STATE
+				failReason,    // FAIL_REASON
+				instance,      // INSTANCE
+				procID,        // PROCESS_ID
+			)
+		}
+		rows = append(rows, row)
 	}
 	return
 }
 
+func getRemainDurationForAnalyzeStatusHelper(
+	sctx sessionctx.Context, startTime interface{},
+	dbName, tableName, partitionName string, processedRows int64) (*time.Duration, float64, int64, error) {
+	var RemainingDuration = time.Duration(0)
+	var percentage = 0.0
+	var totalCnt = int64(0)
+	if startTime != nil {
+		startTime, ok := startTime.(types.Time)
+		if !ok {
+			return nil, percentage, totalCnt, errors.New("invalid start time")
+		}
+		start, err := startTime.GoTime(time.UTC)
+		if err != nil {
+			return nil, percentage, totalCnt, err
+		}
+		duration := time.Now().UTC().Sub(start)
+		is := sessiontxn.GetTxnManager(sctx).GetTxnInfoSchema()
+		tb, err := is.TableByName(model.NewCIStr(dbName), model.NewCIStr(tableName))
+		if err != nil {
+			return nil, percentage, totalCnt, err
+		}
+		var tid int64
+		if partitionName != "" {
+			pt := tb.Meta().GetPartitionInfo()
+			tid = pt.GetPartitionIDByName(partitionName)
+		} else {
+			tid = tb.Meta().ID
+		}
+		if tid > 0 {
+			totalCnt, _ := internalutil.GetApproximateTableCountFromStorage(sctx, tid, dbName, tableName, partitionName)
+			remainline := int64(totalCnt) - processedRows
+			i := float64(remainline) * duration.Seconds() / float64(processedRows)
+			RemainingDuration = time.Duration(i)
+		}
+	}
+	return &RemainingDuration, percentage, totalCnt, nil
+}
+
 // setDataForAnalyzeStatus gets all the analyze jobs.
 func (e *memtableRetriever) setDataForAnalyzeStatus(sctx sessionctx.Context) (err error) {
-	e.rows, err = dataForAnalyzeStatusHelper(sctx)
+	e.rows, err = dataForAnalyzeStatusHelper(sctx, false)
```

Comment:
I think it is also ok to add those fields for `information_schema.analyze_status`.

Reply:
Yes, As the changes are extensive, it may be difficult to review all of them within a single pull request. ```information_schema.analyze_status``` will be in the next PR.